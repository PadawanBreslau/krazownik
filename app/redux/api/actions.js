import {
  LOAD_PAGEDATA_REQUEST,
  LOAD_PAGEDATA_SUCCESS,
  LOAD_PAGEDATA_ERROR,
  SUMBIT_FILE_REQUEST,
  SUBMIT_FILE_SUCCESS,
  SUBMIT_FILE_ERROR,
  SUBMIT_PAGEDATA_REQUEST,
  SUBMIT_PAGEDATA_SUCCESS,
  SUBMIT_PAGEDATA_ERROR,
  UPDATE_DATA_RECORD,
  UPDATE_RELATION_RECORD,
} from './constants';

export default function generateActions(storeName) {
  return {
    loadPageData: (endpoint, callback, errorCallback) => ({
      type: `${LOAD_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      callback,
      errorCallback,
    }),
    loadPageDataSuccess: (payload) => ({
      type: `${LOAD_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
    }),
    loadPageDataError: (error) => ({
      type: `${LOAD_PAGEDATA_ERROR}/${storeName}`,
      error,
    }),
    uploadFileRequest: (endpoint, file, callback) => ({
      type: `${SUMBIT_FILE_REQUEST}/${storeName}`,
      endpoint,
      file,
      callback,
    }),
    uploadFileSuccess: (payload) => ({
      type: `${SUBMIT_FILE_SUCCESS}/${storeName}`,
      payload,
    }),
    uploadFileError: (error) => ({
      type: `${SUBMIT_FILE_ERROR}/${storeName}`,
      error,
    }),
    submitPageData: (
      endpoint,
      method,
      payload,
      successCallbacks,
      searchRedirectUrl,
      errorCallback,
    ) => ({
      type: `${SUBMIT_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      method,
      payload: {
        data: {
          attributes: payload,
        },
      },
      successCallbacks,
      searchRedirectUrl,
      errorCallback,
    }),
    submitPageDataSuccess: (payload) => ({
      type: `${SUBMIT_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
    }),
    submitPageDataError: (error) => ({
      type: `${SUBMIT_PAGEDATA_ERROR}/${storeName}`,
      error,
    }),
    updateRecord: (id, key, value) => ({
      type: `${UPDATE_DATA_RECORD}/${storeName}`,
      id,
      key,
      value,
    }),
    updateRelation: (id, key, value) => ({
      type: `${UPDATE_RELATION_RECORD}/${storeName}`,
      id,
      key,
      value,
    }),
  };
}
