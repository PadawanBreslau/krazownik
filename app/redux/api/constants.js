export const LOAD_PAGEDATA_REQUEST = 'risehigh-frontend/api/LOAD_PAGEDATA';
export const LOAD_PAGEDATA_SUCCESS = 'risehigh-frontend/api/LOAD_PAGEDATA_SUCCESS';
export const LOAD_PAGEDATA_ERROR = 'risehigh-frontend/api/LOAD_PAGEDATA_ERROR';
export const SUMBIT_FILE_REQUEST = 'risehigh-frontend/api/SUMBIT_FILE_REQUEST';
export const SUBMIT_FILE_SUCCESS = 'risehigh-frontend/api/SUBMIT_FILE_SUCCESS';
export const SUBMIT_FILE_ERROR = 'risehigh-frontend/api/SUBMIT_FILE_ERROR';
export const SUBMIT_PAGEDATA_REQUEST = 'risehigh-frontend/api/SUBMIT_PAGEDATA';
export const SUBMIT_PAGEDATA_SUCCESS = 'risehigh-frontend/api/SUBMIT_PAGEDATA_SUCCESS';
export const SUBMIT_PAGEDATA_ERROR = 'risehigh-frontend/api/SUBMIT_PAGEDATA_ERROR';
export const UPDATE_DATA_RECORD = 'risehigh-frontend/api/UPDATE_DATA_RECORD';
export const UPDATE_RELATION_RECORD = 'risehigh-frontend/api/UPDATE_RELATION_RECORD';
