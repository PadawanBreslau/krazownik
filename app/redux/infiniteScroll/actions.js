import {
  CLEAR_PAGEDATA,
  LOAD_PAGEDATA_REQUEST,
  LOAD_PAGEDATA_SUCCESS,
  UPDATE_PAGEDATA_SUCCESS,
  LOAD_PAGEDATA_ERROR,
  SUBMIT_PAGEDATA_REQUEST,
  SUBMIT_PAGEDATA_SUCCESS,
  SUBMIT_PAGEDATA_ERROR,
  UPDATE_DATA_RECORD,
  UPDATE_RELATION_RECORD,
} from './constants';

export default function generateActions(storeName) {
  return {
    clearPageData: () => ({
      type: `${CLEAR_PAGEDATA}/${storeName}`,
    }),
    loadPageData: (endpoint, page) => ({
      type: `${LOAD_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      page,
    }),
    loadPageDataSuccess: (payload) => ({
      type: `${LOAD_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
    }),
    updatePageDataSuccess: (payload) => ({
      type: `${UPDATE_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
    }),
    loadPageDataError: (error) => ({
      type: `${LOAD_PAGEDATA_ERROR}/${storeName}`,
      error,
    }),
    submitPageData: (endpoint, method, payload, successCallbacks, search) => ({
      type: `${SUBMIT_PAGEDATA_REQUEST}/${storeName}`,
      endpoint,
      method,
      payload: {
        data: {
          attributes: payload,
        },
      },
      successCallbacks,
      search,
    }),
    submitPageDataSuccess: (payload, endpoint) => ({
      type: `${SUBMIT_PAGEDATA_SUCCESS}/${storeName}`,
      payload,
      endpoint,
    }),
    submitPageDataError: (error) => ({
      type: `${SUBMIT_PAGEDATA_ERROR}/${storeName}`,
      error,
    }),
    updateRecord: (id, key, value) => ({
      type: `${UPDATE_DATA_RECORD}/${storeName}`,
      id,
      key,
      value,
    }),
    updateRelation: (id, key, value) => ({
      type: `${UPDATE_RELATION_RECORD}/${storeName}`,
      id,
      key,
      value,
    }),
  };
}
