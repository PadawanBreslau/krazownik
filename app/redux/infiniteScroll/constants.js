export const CLEAR_PAGEDATA = 'risehigh-frontend/infiniteScroll/CLEAR_PAGEDATAA';
export const LOAD_PAGEDATA_REQUEST = 'risehigh-frontend/infiniteScroll/LOAD_PAGEDATAA';
export const LOAD_PAGEDATA_SUCCESS = 'risehigh-frontend/infiniteScroll/LOAD_PAGEDATAA_SUCCESS';
export const UPDATE_PAGEDATA_SUCCESS = 'risehigh-frontend/infiniteScroll/UPDATE_PAGEDATA_SUCCESS';
export const LOAD_PAGEDATA_ERROR = 'risehigh-frontend/infiniteScroll/LOAD_PAGEDATA_ERROR';
export const SUBMIT_PAGEDATA_REQUEST = 'risehigh-frontend/infiniteScroll/SUBMIT_PAGEDATA';
export const SUBMIT_PAGEDATA_SUCCESS = 'risehigh-frontend/infiniteScroll/SUBMIT_PAGEDATA_SUCCESS';
export const SUBMIT_PAGEDATA_ERROR = 'risehigh-frontend/infiniteScroll/SUBMIT_PAGEDATA_ERROR';
export const UPDATE_DATA_RECORD = 'risehigh-frontend/infiniteScroll/UPDATE_DATA_RECORD';
export const UPDATE_RELATION_RECORD = 'risehigh-frontend/infiniteScroll/UPDATE_RELATION_RECORD';
