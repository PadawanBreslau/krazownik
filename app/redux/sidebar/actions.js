import {
  LOAD_SIDEBAR,
  LOAD_SIDEBAR_SUCCESS,
  LOAD_SIDEBAR_ERROR,
  CLOSE_SIDEBAR,
  UPDATE_SIDEBAR_DATA,
  UPDATE_SIDEBAR_DATA_RELATION,
} from './constants';

export function loadSidebar(id) {
  return {
    type: LOAD_SIDEBAR,
    id,
  };
}

export function loadSidebarSuccess(payload) {
  return {
    type: LOAD_SIDEBAR_SUCCESS,
    payload,
  };
}

export function loadSidebarError(error) {
  return {
    type: LOAD_SIDEBAR_ERROR,
    error,
  };
}

export function closeSidebar() {
  return {
    type: CLOSE_SIDEBAR,
  };
}

export function updateSidebarData(id, key, value) {
  return {
    type: UPDATE_SIDEBAR_DATA,
    id,
    key,
    value,
  };
}

export function updateSidebarDataRelation(id, key, value) {
  return {
    type: UPDATE_SIDEBAR_DATA_RELATION,
    id,
    key,
    value,
  };
}
