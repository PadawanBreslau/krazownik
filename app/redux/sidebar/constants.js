export const LOAD_SIDEBAR = 'risehigh-frontend/sidebar/LOAD_SIDEBAR';
export const LOAD_SIDEBAR_SUCCESS = 'risehigh-frontend/sidebar/LOAD_SIDEBAR_SUCCESS';
export const LOAD_SIDEBAR_ERROR = 'risehigh-frontend/sidebar/LOAD_SIDEBAR_ERROR';
export const CLOSE_SIDEBAR = 'risehigh-frontend/sidebar/CLOSE_SIDEBAR';
export const UPDATE_SIDEBAR_DATA = 'risehigh-frontend/sidebar/UPDATE_SIDEBAR_DATA';
export const UPDATE_SIDEBAR_DATA_RELATION =
  'risehigh-frontend/sidebar/UPDATE_SIDEBAR_DATA_RELATION';
