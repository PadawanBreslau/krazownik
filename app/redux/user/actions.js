import {
  LOGOUT_REQUEST,
  LOGOUT_SUCCESS,
  LOGOUT_ERROR,
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
} from './constants';

export function LogOut(endpoint) {
  return {
    type: LOGOUT_REQUEST,
    endpoint,
  };
}
export function LogOutSuccess() {
  return {
    type: LOGOUT_SUCCESS,
  };
}
export function LogOutError(error) {
  return {
    type: LOGOUT_ERROR,
    error,
  };
}
export function AuthUser(endpoint, payload, successCallback, isAdmin = false) {
  return {
    type: LOGIN_REQUEST,
    endpoint,
    payload: {
      data: {
        attributes: payload,
      },
    },
    successCallback,
    isAdmin,
  };
}
export function AuthUserSuccess(data, isAdmin) {
  return {
    type: LOGIN_SUCCESS,
    data,
    isAdmin,
  };
}
export function AuthUserError(error) {
  return {
    type: LOGIN_ERROR,
    error,
  };
}
