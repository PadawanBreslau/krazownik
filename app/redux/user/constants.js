export const REGISTER_REQUEST = 'app/user/REGISTER_REQUEST';
export const REGISTER_SUCCESS = 'app/user/REGISTER_SUCCESS';
export const REGISTER_ERROR = 'app/user/REGISTER_ERROR';

export const LOGOUT_REQUEST = 'app/user/LOGOUT_REQUEST';
export const LOGOUT_SUCCESS = 'app/user/LOGOUT_SUCCESS';
export const LOGOUT_ERROR = 'app/user/LOGOUT_ERROR';

export const LOGIN_REQUEST = 'app/user/LOGIN_REQUEST';
export const LOGIN_SUCCESS = 'app/user/LOGIN_SUCCESS';
export const LOGIN_ERROR = 'app/user/LOGIN_ERROR';
