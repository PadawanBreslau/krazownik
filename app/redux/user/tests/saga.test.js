import { put, takeLatest } from 'redux-saga/effects';
import { showUiError } from 'redux/UI/actions';
import { LOGOUT_REQUEST, LOGIN_REQUEST } from '../constants';
import { LogOutSuccess, LogOutError, AuthUserSuccess, AuthUserError } from '../actions';

import defaultSaga, { logOutSaga, logInSaga } from '../saga';

/* eslint-disable redux-saga/yield-effects */
describe('User: Logout Saga', () => {
  let logoutRequestGenerator;
  const exampleResponse = {};
  beforeEach(() => {
    logoutRequestGenerator = logOutSaga({ endpoint: '/endpoint' });
    const putLoadingDescriptor = logoutRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = logoutRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
  it('should dispatch the LogOutSuccess action if it requests the data successfully', () => {
    const putDescriptor = logoutRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(LogOutSuccess(exampleResponse)));
  });
  it('should call the LogOutError action if the response errors', () => {
    const response = { some: 'Logout error' };
    const putShowErrorDescriptor = logoutRequestGenerator.throw(response).value;
    expect(putShowErrorDescriptor).toEqual(put(showUiError(response)));
    const putErrorDescriptor = logoutRequestGenerator.next().value;
    expect(putErrorDescriptor).toEqual(put(LogOutError(response)));
  });
});
describe('User: Login Saga', () => {
  let logoutRequestGenerator;
  const exampleResponse = {};
  const callback = jest.fn();
  beforeEach(() => {
    logoutRequestGenerator = logInSaga({
      endpoint: '/endpoint',
      payload: {},
      successCallback: callback,
    });
    const putLoadingDescriptor = logoutRequestGenerator.next().value;
    expect(putLoadingDescriptor).toMatchSnapshot();
    const callDescriptor = logoutRequestGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
  it('should dispatch the LogInSuccess action if it requests the data successfully', () => {
    const putDescriptor = logoutRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(AuthUserSuccess(exampleResponse)));
  });
  it('should do the callback', () => {
    const putDescriptor = logoutRequestGenerator.next(exampleResponse).value;
    expect(putDescriptor).toEqual(put(AuthUserSuccess(exampleResponse)));
    const endLoadingDescriptor = logoutRequestGenerator.next(exampleResponse).value;
    expect(endLoadingDescriptor).toMatchSnapshot();
    const putCallbackDescriptor = logoutRequestGenerator.next(exampleResponse).value;
    expect(putCallbackDescriptor).toEqual(put(callback));
  });
  it('should call the LogInError action if the response errors', () => {
    const response = { some: 'Logout error' };
    const putShowErrorDescriptor = logoutRequestGenerator.throw(response).value;
    expect(putShowErrorDescriptor).toEqual(put(showUiError(response)));
    const putErrorDescriptor = logoutRequestGenerator.next().value;
    expect(putErrorDescriptor).toEqual(put(AuthUserError(response)));
  });
});

describe('User: defaultSaga watcher', () => {
  const watcher = defaultSaga();
  it('should start task to watch for LOGOUT_REQUEST action', () => {
    const takeLatestDescriptor = watcher.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOGOUT_REQUEST, logOutSaga));
  });
  it('should start task to watch for LOGIN_REQUEST action', () => {
    const takeLatestDescriptor = watcher.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOGIN_REQUEST, logInSaga));
  });
});
