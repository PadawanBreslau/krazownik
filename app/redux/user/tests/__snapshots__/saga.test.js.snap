// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User: Login Saga should call the LogInError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Login Saga should call the LogInError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/endpoint",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`User: Login Saga should dispatch the LogInSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Login Saga should dispatch the LogInSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/endpoint",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`User: Login Saga should do the callback 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Login Saga should do the callback 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/endpoint",
      Object {},
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`User: Login Saga should do the callback 3`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/HIDE_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Logout Saga should call the LogOutError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Logout Saga should call the LogOutError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/endpoint",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`User: Logout Saga should dispatch the LogOutSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "app/Messages/SHOW_LOADING",
    },
    "channel": null,
  },
}
`;

exports[`User: Logout Saga should dispatch the LogOutSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/endpoint",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
